src/aspire/basis/polar_2d.py

```diff
@@ -0,0 +1,111 @@
+        self.sz_prod = self.sz[0] * self.sz[1]
+
+        # precompute the basis functions in 2D grids
+        self.freqs = self._precomp()
+
+    def _precomp(self):
+        """
+        Precomute the basis functions on a polar Fourier grid
+        """
+        omega0 = 2 * np.pi / (2 * self.nrad - 1)
+        dtheta = 2 * np.pi / self.ntheta
+
+        # Note: The size of ntheta can be reduced by half if only evaluate_t function is needed as
+        # in the Matlab implementation, np.zeros((2, self.nrad * self.ntheta//2)).
+        freqs = np.zeros((2, self.nrad * self.ntheta))
+        for i in range(self.ntheta):
+            freqs[0, i * self.nrad: (i + 1) * self.nrad] = np.arange(self.nrad) * np.sin(i * dtheta)
+            freqs[1, i * self.nrad: (i + 1) * self.nrad] = np.arange(self.nrad) * np.cos(i * dtheta)
+
+        freqs *= omega0
+        return freqs
+
+    def evaluate_t(self, x):
+        """
+        Evaluate coefficient in polar Fourier basis from those in standard 2D coordinate basis
+
+        :param x: The coefficient array in the standard 2D coordinate basis to be
+            evaluated. The first two dimensions must equal `self.sz`.
+        :return v: The evaluation of the coefficient array `v` in the polar Fourier basis.
+            This is an array of vectors whose first two dimensions are `self.nrad` and
+            `self.ntheta` and whose remaining dimensions correspond to higher dimensions of `x`.
+        """
+        # ensure the first two dimensions with size of self.sz
+        x, sz_roll = unroll_dim(x, self.ndim + 1)
+        x = m_reshape(x, (self.sz[0], self.sz[1], -1))
+        nimgs = x.shape[2]
+
+        # finufftpy require it to be aligned in fortran order
+        half_size = self.nrad * self.ntheta // 2
+        pf = np.empty((half_size, nimgs), dtype='complex128', order='F')
+        finufftpy.nufft2d2many(self.freqs[0, 0:half_size], self.freqs[1, 0:half_size], pf, 1, 1e-15, x)
+        pf = m_reshape(pf, (self.nrad, self.ntheta // 2, nimgs))
+        v = np.concatenate((pf, pf.conj()), axis=1).copy()
+
+        # return v coefficients with the first dimension of self.count
+        v = roll_dim(v, sz_roll)
+        return v
+
+    def expand(self, x):
+        """
+        Obtain coefficients in polar Fourier basis from those in standard 2D coordinate basis
+
+        This is the same function to `evaluate_t` for a consistent implementation
+        with other expanding methods.
+
+        :param x: The coefficient array in the standard 2D coordinate basis to be
+            evaluated. The first two dimensions must equal `self.sz`.
+        :return v: The evaluation of the coefficient array `v` in the polar Fourier basis.
+            This is an array of vectors whose first two dimensions are `self.nrad` and
+            `self.ntheta` and whose remaining dimensions correspond to higher dimensions of `x`.
+        """
+        return self.evaluate_t(x)
```

Comment:
This isn't really valid. The polar Fourier transform is not an orthogonal basis, so the inverse doesn't equal the transpose.

Reply:
For simplicity, I will remove it.

Reply:
removed.